### PR TITLE
Ga updates

### PR DESCRIPTION
**Pull request by wright**
Genetic Algorithm update adds k-point crossover (on by default) and optional Gray code binary representation (off by default).

### CODE REVIEW

File: openmdao/drivers/genetic_algorithm_driver.py [hunk 1]
```diff
@@ -215,6 +218,8 @@ def run(self):
         ga = self._ga
 
         ga.elite = self.options['elitism']
+        ga.grayCode = self.options['gray']
+        ga.crossBits = self.options['cross_bits']
```
Comment (Kenneth-T-Moore) [inline openmdao/drivers/genetic_algorithm_driver.py]: With the coding and PEP styles we use, we don't usually use camel case for variable names or method names.  I would use gray_code and cross_bits instead.

File: openmdao/drivers/genetic_algorithm_driver.py [hunk 2]
```diff
@@ -807,11 +827,54 @@ def encode(self, x, vlb, vub, bits):
...
         x = np.round((x - vlb) / interval).astype(np.int)
         byte_str = [("0" * b + bin(i)[2:])[-b:] for i, b in zip(x, bits)]
-        return np.array([int(c) for s in byte_str for c in s])
+        result = np.array([int(c) for s in byte_str for c in s])
+        if self.grayCode:
+            result = self.toGray(result)
+        return result
+
+    def toGray(self, g):
+        """
```
Comment (Kenneth-T-Moore) [inline openmdao/drivers/genetic_algorithm_driver.py]: similarly to_gray and from_gray

**Comment by robfalck** [comment]
Could you please include a brief description of Gray coding in the desc string for that option?  That way our documentation generator will at least include that when it lists out the options table.

http://openmdao.org/twodocs/versions/latest/features/building_blocks/drivers/genetic_algorithm.html

File: openmdao/drivers/genetic_algorithm_driver.py [hunk 3]
```diff
@@ -807,11 +830,55 @@ def encode(self, x, vlb, vub, bits):
...
+        result = np.array([int(c) for s in byte_str for c in s])
+        if self.gray_code:
+            result = self.to_gray(result)
+        return result
+
+    def to_gray(self, g):
+        """
+        Convert a Gray coded binary array representing a single population member
+        to normal binary coding.
+
```
Comment (Kenneth-T-Moore) [inline openmdao/drivers/genetic_algorithm_driver.py]: Docstring descriptions seem tot be reversed. This one is to_gray, but the string says convert from gray.

File: openmdao/drivers/genetic_algorithm_driver.py (hunk 1 above)
Comment (wright) [reply openmdao/drivers/genetic_algorithm_driver.py]: fixed

File: openmdao/drivers/genetic_algorithm_driver.py (hunk 2 above)
Comment (wright) [reply openmdao/drivers/genetic_algorithm_driver.py]: fixed

File: openmdao/drivers/genetic_algorithm_driver.py (hunk 3 above)
Comment (wright) [reply openmdao/drivers/genetic_algorithm_driver.py]: fixed

**Pull request merged**